vault/quotas/quotas_rate_limit.go

```diff
@@ -150,10 +165,68 @@ func (rlq *RateLimitQuota) initialize(logger log.Logger, ms *metricsutil.Cluster
 	}
 
 	rlq.store = rlStore
+	rlq.blockedClients = make(map[string]time.Time)
+
+	if rlq.BlockInterval > 0 && !rlq.purgeBlocked {
+		rlq.purgeBlocked = true
+		rlq.closePurgeBlockedCh = make(chan struct{})
+		go rlq.purgeBlockedClients()
+	}
 
 	return nil
 }
 
+// purgeBlockedClients performs a blocking process where every purgeInterval
+// duration, we look at all blocked clients to potentially remove from the blocked
+// clients map.
+//
+// A blocked client will only be removed if the current time minus the time the
+// client was blocked at is greater than or equal to the block duration. The loop
+// will continue to run indefinitely until a value is	sent on the closePurgeBlockedCh
+// in which we stop the ticker and return.
+func (rlq *RateLimitQuota) purgeBlockedClients() {
+	rlq.lock.RLock()
+	ticker := time.NewTicker(rlq.purgeInterval)
+	rlq.lock.RUnlock()
+
+	for {
+		select {
+		case t := <-ticker.C:
+			rlq.lock.Lock()
```

Comment:
Technically, it could be unbounded, but I can't imagine it'll be that large in practice. We could use a `sync.Map`, but doesn't that come with relatively significant performance bottlenecks. I have no strong opinion here.

Reply:
I don't know for sure-- we could measure it-- but my guess is that it would be less of a bottleneck than every request covered by a quota contending for the same lock.

However, making that a RLock instead of a Lock might be sufficient, I'll put a comment on that? But that introduces the same problem because we need to switch to a write-lock in order to update blockedClients.